scripts/data_prep/README.md

```diff
@@ -12,21 +12,48 @@ Using the `convert_dataset_hf.py` script...
 python convert_dataset_hf.py \
   --dataset c4 --data_subset en \
   --out_root my-copy-c4 --splits train_small val_small \
-  --concat_tokens 2048 --tokenizer EleutherAI/gpt-neox-20b --eos_text '<|endoftext|>'
+  --concat_tokens 2048 --tokenizer EleutherAI/gpt-neox-20b --eos_text '<|endoftext|>' \
+  --compression zstd
 ```
 
 Using the `convert_dataset_json.py` script...
 
 <!--pytest.mark.skip-->
 ```bash
 # Convert C4 dataset to StreamingDataset format
-python convert_dataset_hf.py \
+python convert_dataset_json.py \
   --path ./example_data/arxiv.json \
   --out_root my-copy-arxiv --split train \
-  --concat_tokens 2048 --tokenizer EleutherAI/gpt-neox-20b --eos_text '<|endoftext|>'
+  --concat_tokens 2048 --tokenizer EleutherAI/gpt-neox-20b --eos_text '<|endoftext|>' \
+  --compression zstd
 ```
 
 Where `--path` can be a single json file, or a folder containing json files, and split the intended split (hf defaults to train).
 
 ## Converting a finetuning dataset
-Using the `convert_finetuning_dataset.py` script...
+Using the `convert_finetuning_dataset.py` script you can run a commandsuch as:
+
+```bash
+python convert_finetuning_dataset.py --dataset ""Muennighoff/P3"" \
+--splits "train" "validation" \
+--preprocessor "your_preprocessing_function"\
+ --out_root "/path/to/your/output_directory"
+```
+
+This example assumes:
+
+- You are running the script in the terminal or command prompt where `python` command is recognized.
+- `"Muennighoff/P3"` is the dataset you want to convert. Substitute "Muennighoff/P3" with the name or path of your dataset.
+- `train` and `validation` are the splits of the dataset to convert.
+- `llmfoundry.data.finetuning.tasks:p3_preprocessing_function` is a string that provides the name or import path of the function used to preprocess the dataset. Substitute it with your actual preprocessor.
```

Comment:
i think there are more details on how to define a preprocessing function somewhere else, maybe point to those?

Reply:
Reviewing this had led me to catch some issues with preprocessing. I'll address that in a separate PR, along with other improvements.

Reply:
@dakinggg  I updated with a link to relevant files, but unless I missed it I can't find pointers on how to define the function (other than the code)